File: java-queue/src/com/erdemkeren/Queue.java
package com.erdemkeren;

import java.util.LinkedList;

public class Queue {
    private LinkedList queue = new LinkedList();

    public boolean isEmpty() { return queue.isEmpty(); }

    public int size()
    {
        return queue.size();
    }

    public void enqueue(int val) { queue.addLast(val); }

    public int dequeue() {
        return (int) queue.pop();
    }

    public int peek() {
        return (int) queue.get(0);
    }
}

File: csharp-queue/csharp-queue/csharp-queue/Queue.cs
using System.Linq;
using System.Collections.Generic;

namespace csharp_queue
{
    public class Queue
    {
        private readonly LinkedList<int> _queue = new LinkedList<int>();

        public int Size() => _queue.Count;

        public bool IsEmpty() => _queue.Any();

        public void Enqueue(int val) => _queue.AddLast(val);

        public int Dequeue()
        {
            var element = _queue.First();
            _queue.RemoveFirst();
            
            return element;
        }

        public int Peek() => _queue.ElementAt(0); 
    }
}